### PR TITLE
Make sure that NeoStores closes all successfully initialised stores if initialisation of a store fails.

### PR DESCRIPTION
**Pull request by chrisvest**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 1]
```diff
@@ -142,9 +142,24 @@ public boolean test( StoreType type )
...
         stores = new Object[StoreType.values().length];
-        for ( StoreType type : storeTypes )
+        try
+        {
+            for ( StoreType type : storeTypes )
+            {
+                getOrCreateStore( type );
+            }
+        }
+        catch ( Exception initException )
```
Comment (klaren) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: `RuntimeException`

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 2]
```diff
@@ -142,9 +142,24 @@ public boolean test( StoreType type )
...
         {
-            getOrCreateStore( type );
+            try
+            {
+                close();
+            }
+            catch ( Exception closeException )
```
Comment (klaren) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: Same

**Review by klaren — APPROVED** [review]
Two small changes and your good to go. I'm pre-approving this since the requested changes are so small.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java (hunk 1 above)
Comment (MishaDemianenko) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: can IOUtils.closeAll help here?

Comment (chrisvest) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: It's a slightly different situation than what IOUtil.closeAll is good for. We're only closing one thing, kind of, and only if the initialisation fails.

**Pull request merged**